CacheService/docker_context/CacheHandler.py

```diff
@@ -224,4 +267,117 @@ async def getQuote(self, trans_num, user_id, stock_id):
             )
             return result, status
 
+    async def getBulkQuote(self, user_ids, stock_ids, transaction_nums):
+        results = []
+        while stock_ids:
+            user_id = user_ids.pop(0)
+            stock_id = stock_ids.pop(0)
+            transaction_num = transaction_nums.pop(0)
+
+            check = await self.RedisHandler.rExists(stock_id)
+            if check:
+                quote = await self.RedisHandler.rGet(stock_id)
+                then = float(quote['time'])
+                now = currentTime()
+                difference = (now - then)
+                if difference <= self._QUOTE_CACHE_TIME_LIMIT_SEC:
+                    logger.debug(f'{__name__} - Cache hit while getting stock {stock_id}')
+                    results.append({'stock_id': stock_id, 'price': quote['price']})
+                    continue
+
+            result, status = await self.LegacyStock.getQuote(transaction_num, user_id, stock_id)
+            if status == 200:
+                await self.RedisHandler.rSet(stock_id,
+                                             {'stock_id': stock_id, 'price': result['price'], 'time': currentTime()})
+                result['stock_id'] = stock_id
+                results.append(result)
+            else:
+                # not sure how we want to handle this, so I'm just using this from quote.
+                logger.error(
+                    f'{__name__} - Cannot get quote for {transaction_num}, {user_id}, {stock_id} due to an error.'
+                    f' {status} --> {result}')
+                self.audit.handleError(
+                    trans_num=transaction_num,
+                    command='QUOTE',
+                    error_msg=f'Error getting quote - {result} ({status})',
+                    user_id=user_id,
+                    stock_symbol=stock_id
+                )
+        # not sure what we want to do if the error above happens.
+        return results, 200
+
+    async def setBuyAmount(self, data):
+        result, status = await self.client.postRequest(f'{self.dbmURL}/triggers/buy/set/amount', data)
+        if status == 200:
+            await self.cacheAccountTransaction(data["user_id"])
+        return result, status
+
+    async def setSellAmount(self, data):
+        result, status = await self.client.postRequest(f'{self.dbmURL}/triggers/sell/set/amount', data)
+        if status == 200:
+            await self.cacheStockTransaction(data['user_id'], data['stock_symbol'])
+        return result, status
+
+    async def executeTriggers(self, data):
```

Comment:
may be interesting having this being executed in a ProcessPoolExecutor and running this method on a separate process since it could potentially take a long time to execute and we don't want it constantly taking control in the event loop

Reply:
Give the latest commit a look